### PR TITLE
Put in some guardrails against overly large confusion matrices

### PR DESCRIPTION
**Pull request by jamie256**
## Description

After investigating some larger profile sizes we saw the confusion matrix is currently not limited in size and is likely a mistake when created with more than 256 labels. We need further work to efficiently handle larger numbers of labels, so this change is a simple mitigation until we have that support.

### General Checklist

* [x] Tests added for this feature/bug
      if it was a bug, test must cover it.
* [x] Conform by the style guides, by using formatter
* [ ] Documentation updated
* [ ] (optional) Please add a label to your PR

### CODE REVIEW

File: tests/unit/core/metrics/test_confusion_matrix.py [hunk 1]
```diff
@@ -10,6 +12,36 @@
 from whylogs.proto import ScoreMatrixMessage
 
 
+def _generateTestTargetsAndPredictions(label_size: int, output_length, rand_ratio: float):
+    labels = [i for i in range(label_size)]
```
Comment (richard-rogers) [inline tests/unit/core/metrics/test_confusion_matrix.py]: why not just `labels = range(label_size)` ?

File: tests/unit/core/metrics/test_confusion_matrix.py [hunk 2]
```diff
@@ -10,6 +12,36 @@
...
+    return targets, predictions
+
+
+def _generateTestTargetsAndPredictionsForEachLabel(label_size: int, output_length, rand_ratio: float):
+    labels = [i for i in range(label_size)]
+    targets = []
+    predictions = []
+
+    for i in range(output_length):
+        label = labels[i % label_size]
```
Comment (richard-rogers) [inline tests/unit/core/metrics/test_confusion_matrix.py:21]: Pass label selector in as a callable so these two can be unified?

File: tests/unit/core/metrics/test_confusion_matrix.py (hunk 1 above)
Comment (jamie256) [reply tests/unit/core/metrics/test_confusion_matrix.py]: yeah better to simplify, done.

File: tests/unit/core/metrics/test_confusion_matrix.py (hunk 2 above)
Comment (jamie256) [reply tests/unit/core/metrics/test_confusion_matrix.py:21]: I removed one of the methods for now, but good suggestion.

File: tests/unit/core/metrics/test_confusion_matrix.py [hunk 3]
```diff
@@ -157,6 +175,29 @@ def test_confusion_matrix_to_protobuf():
...
 
 
+def test_over_threshold_confusion_matrix():
+    with pytest.raises(ValueError):
+        ConfusionMatrix(range(257))
+
+
+def test_merged_labels_over_threshold_confusion_matrix():
+    labels = range(200)
+    more_labels = [j for j in range(201, 300)]
```
Comment (richard-rogers) [inline tests/unit/core/metrics/test_confusion_matrix.py]: Why not just `range(201, 300)` ?

File: tests/unit/core/metrics/test_confusion_matrix.py [hunk 4]
```diff
@@ -10,6 +12,22 @@
 from whylogs.proto import ScoreMatrixMessage
 
 
+def _generateTestTargetsAndPredictionsForEachLabel(label_size: int, output_length, rand_ratio: float):
+    # labels = [i for i in range(label_size)]
```
Comment (richard-rogers) [inline tests/unit/core/metrics/test_confusion_matrix.py]: remove comment?

Comment (jamie256) [reply tests/unit/core/metrics/test_confusion_matrix.py]: nice catch! thanks.

File: tests/unit/core/metrics/test_confusion_matrix.py (hunk 3 above)
Comment (jamie256) [reply tests/unit/core/metrics/test_confusion_matrix.py]: true, done.

**Review by richard-rogers — APPROVED** [review]

**Pull request merged**